browser/extensions/brave_tor_client_updater.cc

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/extensions/brave_tor_client_updater.h"
+
+#include "base/files/file_enumerator.h"
+#include "base/files/file_path.h"
+#include "base/task_scheduler/post_task.h"
+#include "third_party/re2/src/re2/re2.h"
+
+namespace extensions {
+
+std::string BraveTorClientUpdater::g_tor_client_component_id_(
+    kTorClientComponentId);
+std::string BraveTorClientUpdater::g_tor_client_component_base64_public_key_(
+    kTorClientComponentBase64PublicKey);
+
+BraveTorClientUpdater::BraveTorClientUpdater()
+    : task_runner_(
+          base::CreateSequencedTaskRunnerWithTraits({base::MayBlock()})),
+      registered_(false) {
+}
+
+BraveTorClientUpdater::~BraveTorClientUpdater() {
+}
+
+void BraveTorClientUpdater::Register() {
+  if (registered_)
+    return;
+
+  BraveComponentExtension::Register(kTorClientComponentName,
+                                    g_tor_client_component_id_,
+                                    g_tor_client_component_base64_public_key_);
+  registered_ = true;
+}
+
+base::FilePath BraveTorClientUpdater::GetExecutablePath() const {
+  return executable_path_;
+}
+
+void BraveTorClientUpdater::InitExecutablePath(
+    const base::FilePath& install_dir) {
+  base::FileEnumerator traversal(install_dir, false,
```

Comment:
This should be flagged for security audit. We shouldn't be using a wildcard here to find files, we need to look for a specific file and match the contents against a checksum cc @diracdeltas

Reply:
Ah, good point. I implemented it like that because the Tor client embeds versioning and target platform information in its executable name (example: `tor-0.3.3.8-win32-brave-5`) and I didn't want to tightly couple the updater with a specific Tor client. I think one thing we can do here is to change the name of the Tor client to something a bit more generic when packaging it in the extension (e.g., `brave-tor-client` or something like that).

A checksum makes sense, but I think it would mean that we couldn't update the Tor client without also updating Brave. I think that's at least part of the problem the extension was meant to address.

cc: @riastradh-brave and @bbondy for any other thoughts here.

Reply:
we do validate the crx so the checksum can be included along with the exe

Reply:
Thank you for flagging,  I am assigning to @riastradh-brave for retroactive security review.

Reply:
I would be perfectly fine with hard-coding a hash -- that's what we do currently.  As I understand it, the only real specific motivation we have for doing a separate extension is to avoid antivirus software that is unhappy if Brave is bundled with a tor executable -- but even then, I'm not sure have any reports of antivirus software complaining that it's _there_; all the reports I'm aware of are complaints that it got _executed_.

Reply:
Generally, we put out updates more frequently than tor does anyway.

Reply:
I see what's going on here.  @emerick Can we just use a fixed pathname at a fixed location for the executable file?  The only reason the file that is published via S3 has a different name is that we are publishing several different files, one for each platform.

Reply:
@riastradh-brave Yeah, I'll update the packager to use a more generic name for the executable and then we'll just look specifically for that.